Build failure: lgogdownloader

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

1. Run `nix build nixpkgs#lgogdownloader`
2. Get build error

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/326820901

### Relevant log output

```console
[ 91%] Linking CXX executable lgogdownloader
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::sortGalaxyProductBuilds(Json::Value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(Json::Value const&, Json::Value const&)#1}::operator()(Json::Value const&, Json::Value const&) const [clone .isra.0]':
downloader.cpp:(.text+0x2738): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x2753): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `void std::__adjust_heap<__gnu_cxx::__normal_iterator<Json::Value*, std::vector<Json::Value, std::allocator<Json::Value> > >, long, Json::Value, __gnu_cxx::__ops::_Iter_comp_iter<Downloader::sortGalaxyProductBuilds(Json::Value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(Json::Value const&, Json::Value const&)#1}> >(__gnu_cxx::__normal_iterator<Json::Value*, std::vector<Json::Value, std::allocator<Json::Value> > >, long, long, Json::Value, __gnu_cxx::__ops::_Iter_comp_iter<Downloader::sortGalaxyProductBuilds(Json::Value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(Json::Value const&, Json::Value const&)#1}>)':
downloader.cpp:(.text+0x2a63): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x2a85): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x2c72): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o:downloader.cpp:(.text+0x2c8c): more undefined references to `Json::Value::operator[](char const*) const' follow
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::checkNotifications()':
downloader.cpp:(.text+0x8405): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8438): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8447): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8485): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8494): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x84d2): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o:downloader.cpp:(.text+0x84e1): more undefined references to `Json::Value::operator[](char const*)' follow
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::clearUpdateNotifications()':
downloader.cpp:(.text+0x883f): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8856): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8865): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::galaxyGetBuildIndexWithBuildId(Json::Value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
downloader.cpp:(.text+0x8c6f): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8c8e): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x8ca7): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::getChangelogFromJSON[abi:cxx11](Json::Value const&)':
downloader.cpp:(.text+0x96f9): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x9786): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x97e5): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x9800): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::sortGalaxyProductBuilds(Json::Value, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
downloader.cpp:(.text+0x9fe2): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa347): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa365): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa471): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa4a0): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa4b6): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa617): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa635): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xa7b3): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xaa87): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xaac0): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xaad6): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `std::_Function_handler<void (cloudSaveFile&), Downloader::deleteCloudSavesById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(cloudSaveFile&)#1}>::_M_invoke(std::_Any_data const&, cloudSaveFile&)':
downloader.cpp:(.text+0xe273): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xe2e7): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xe670): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xe6a8): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::init()':
downloader.cpp:(.text+0xef07): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0xef1f): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::Downloader()':
downloader.cpp:(.text+0x1327b): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x132de): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x13308): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x13363): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1340d): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1343c): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x134fb): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1351a): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::galaxyListCDNsById(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
downloader.cpp:(.text+0x14808): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x14822): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o:downloader.cpp:(.text+0x14846): more undefined references to `Json::Value::operator[](char const*)' follow
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::getGalaxyInstallDirectory[abi:cxx11](galaxyAPI*, Json::Value const&)':
downloader.cpp:(.text+0x15ada): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x15b40): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x15f54): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x15f76): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x15fb0): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1661f): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::cloudSaveLocations(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
downloader.cpp:(.text+0x16a97): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o:downloader.cpp:(.text+0x16ab1): more undefined references to `Json::Value::operator[](char const*)' follow
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::cloudSaveLocations(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
downloader.cpp:(.text+0x16cd7): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x16f76): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x172cf): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x172fe): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1730d): undefined reference to `Json::Value::operator[](std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1731c): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x17354): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1771b): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1776a): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x1779e): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x178ff): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x17a97): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/downloader.cpp.o: in function `Downloader::galaxyGetDepotItemVectorFromJson(Json::Value const&, unsigned int const&)':
downloader.cpp:(.text+0x21300): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x2143b): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x21470): undefined reference to `Json::Value::operator[](char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x2174c): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: downloader.cpp:(.text+0x2179c): undefined reference to `Json::Value::operator[](char const*) const'

...

/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x11c74): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x11ca2): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x11cc9): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x11ea7): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x11f11): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x124ec): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x12509): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x12535): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x1279f): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x128cf): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x12bd3): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x12bf2): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/website.cpp.o:website.cpp:(.text+0x1303e): more undefined references to `Json::Value::operator[](char const*)' follow
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/website.cpp.o: in function `Website::getWishlistItems()':
website.cpp:(.text+0x134fc): undefined reference to `Json::Value::isMember(char const*) const'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x13513): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x13536): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x13545): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x135d9): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: website.cpp:(.text+0x135e8): undefined reference to `Json::Value::operator[](char const*)'
/nix/store/p2vkw5s89ff1fs2d2rxqxiqil9s0jpcm-binutils-2.46/bin/ld.bfd: CMakeFiles/lgogdownloader.dir/src/website.cpp.o:website.cpp:(.text+0x1360e): more undefined references to `Json::Value::operator[](char const*)' follow
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/lgogdownloader.dir/build.make:258: lgogdownloader] Error 1
make[1]: *** [CMakeFiles/Makefile2:109: CMakeFiles/lgogdownloader.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 7.0.0-cachyos-lto, NixOS, 26.05 (Yarara), 26.05.20260422.0726a0e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.34.6`
 - nixpkgs: `/nix/store/75bkaivfbwq3x8cs7155hag7hs1chjcx-source`

### Notify maintainers

@0x4A6F

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/lg/lgogdownloader/package.nix
+++ b/pkgs/by-name/lg/lgogdownloader/package.nix
@@ -30,6 +30,11 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-dVEV2smZxB6+Utm9FApiFydAS3hLm4y9YZja1B/PiEk=";
   };
 
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "set_property(TARGET \''${PROJECT_NAME} PROPERTY CXX_STANDARD 11)" "set_property(TARGET \''${PROJECT_NAME} PROPERTY CXX_STANDARD 17)"
+  '';
+
   nativeBuildInputs = [
     cmake
     pkg-config
@@ -52,7 +57,11 @@ stdenv.mkDerivation (finalAttrs: {
     libsForQt5.qtwebengine
   ];
 
-  cmakeFlags = lib.optional enableGui "-DUSE_QT_GUI=ON";
+  cmakeFlags = [
+    (lib.cmakeBool "USE_QT_GUI" enableGui)
+    (lib.cmakeFeature "CMAKE_CXX_STANDARD" "17")
+    (lib.cmakeFeature "CMAKE_CXX_FLAGS" "-DJSONCPP_HAS_STRING_VIEW=1")
+  ];
 
   passthru.tests = {
     version = testers.testVersion { package = finalAttrs.finalPackage; };

